app/lib/account/consent_backend.dart

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:client_data/account_api.dart' as api;
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:retry/retry.dart';
+
+import '../frontend/email_sender.dart';
+import '../shared/email.dart' show createInviteEmail;
+import '../shared/exceptions.dart';
+import '../shared/urls.dart';
+
+import 'backend.dart';
+import 'models.dart';
+
+final _logger = Logger('pub.consent.backend');
+
+/// Sets the consent backend service.
+void registerConsentBackend(ConsentBackend backend) =>
+    ss.register(#_consentBackend, backend);
+
+/// The active consent backend service.
+ConsentBackend get consentBackend =>
+    ss.lookup(#_consentBackend) as ConsentBackend;
+
+/// Represents the backend for the consent handling and authentication.
+class ConsentBackend {
+  final DatastoreDB _db;
+  final _actions = <String, ConsentAction>{};
+
+  ConsentBackend(this._db);
+
+  /// Returns the consent details.
+  Future<api.Consent> getConsent(String consentId) async {
+    return await withAuthenticatedUser((user) async {
+      final key = _db.emptyKey
+          .append(User, id: user.userId)
+          .append(Consent, id: consentId);
+      final c = (await _db.lookup<Consent>([key])).single;
+      if (c == null) {
+        throw NotFoundException.resource('consent: $consentId');
+      }
+      return api.Consent(html: c.descriptionHtml);
+    });
+  }
+
+  /// Resolves the consent.
+  Future<api.ConsentResult> resolveConsent(
+      String consentId, api.ConsentResult result) async {
+    return await withAuthenticatedUser((user) async {
+      final key = _db.emptyKey
+          .append(User, id: user.userId)
+          .append(Consent, id: consentId);
+      final c = (await _db.lookup<Consent>([key])).single;
+      if (c == null) {
+        throw NotFoundException.resource('consent: $consentId');
+      }
+      if (result.granted ?? false) {
```

Comment:
The `ConsentResult` does not specify the value of `granted` if left as `null`, this might be an oversight, since the property doesn't have documentation.

But it's probably better to do:
`InvalidInputException.checkNotNull(result.granted, 'granted');`